pkg/apis/app/v2/kieapp_types.go

```diff
@@ -348,21 +352,28 @@ const (
 // DatabaseObject Defines how a KieServer will manage and create a new Database
 // or connect to an existing one
 type DatabaseObject struct {
-	Type             DatabaseType            `json:"type,omitempty"`
-	Size             string                  `json:"size,omitempty"`
-	StorageClassName string                  `json:"storageClassName,omitempty"`
-	ExternalConfig   *ExternalDatabaseObject `json:"externalConfig,omitempty"`
+	InternalDatabaseObject `json:",inline"`
+	ExternalConfig         *ExternalDatabaseObject `json:"externalConfig,omitempty"`
 }
 
-// ExternalDatabaseObject configuration definition of an external database
-type ExternalDatabaseObject struct {
+// ProcessMigrationDatabaseObject Defines how a Process Migration server will manage
+// and create a new Database or connect to an existing one
+type ProcessMigrationDatabaseObject struct {
```

Comment:
Because the `ExternalConfig` of `DatabaseObject` has `Dialect`, `Name`, `Host`, `Port`, `NonXA`. I don't think these are applicable to the data source configuration for process migration.

Reply:
After reading the documentation and scripts in KIE Server it seems these properties won't be needed because they are used in mysql and postgres but it is possible to configure the datasources without them.
In EAP they are set as datasource properties so in Thorntail it should be set like this:
`thorntail.datasources.xa-data-sources.KEY.xa-datasource-properties.ServerName.value = Host`
`thorntail.datasources.xa-data-sources.KEY.xa-datasource-properties.PortNumber.value = Port`
The dialect is used in the kie server scripts to decide what to require in terms of properties depending on the value so no need to use it here either.